core/src/main/java/com/opensymphony/xwork2/util/AbstractLocalizedTextProvider.java

```diff
@@ -61,7 +59,7 @@
 
     private final ConcurrentMap<MessageFormatKey, MessageFormat> messageFormats = new ConcurrentHashMap<>();
     private final ConcurrentMap<Integer, List<String>> classLoaderMap = new ConcurrentHashMap<>();
-    private final Set<String> missingBundles = Collections.synchronizedSet(new HashSet<String>());
+    private final ConcurrentMap<String,Boolean> missingBundles = new ConcurrentHashMap<>();
```

Comment:
Minor formatting suggestion:  Add a space before `Boolean` for: `ConcurrentMap<String, Boolean>`.